x-pack/plugins/files/server/file_service/internal_file_service.ts

```diff
@@ -223,4 +228,47 @@ export class InternalFileService {
     });
     return result.saved_objects.map((so) => toJSON(so.id, so.attributes));
   }
+
+  public async getUsageMetrics(): Promise<FilesMetrics> {
+    let pit: undefined | SavedObjectsOpenPointInTimeResponse;
+    try {
+      pit = await this.soClient.openPointInTimeForType(this.savedObjectType);
+      const deleted: FileStatus = 'DELETED';
+      const { aggregations } = await this.soClient.find<
+        FileSavedObjectAttributes,
+        { size: AggregationsSumAggregate }
+      >({
+        type: this.savedObjectType,
+        pit,
+        filter: `NOT ${this.savedObjectType}.attributes.Status: ${deleted}`,
```

Comment:
Maybe we could also aggregate by the `Status` field and show the number of files in each state.

Reply:
Also could aggreagate by mime type main type, if we had such a field, and show what type of files are stored. Maybe also by extension.